src/wire/sixlowpan.rs

```diff
@@ -0,0 +1,1720 @@
+use crate::wire::ieee802154::Address as LlAddress;
+use crate::wire::ipv6;
+use crate::wire::IpProtocol;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum NextHeader {
+    Compressed,
+    Uncompressed(IpProtocol),
+}
+
+/// A wrapper around the address provided in the 6LoWPAN_IPHC header.
+/// This requires some context to convert it the an IPv6 address in some cases.
+/// For 802.15.4 the context are the short/extended addresses.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum Address<'a> {
+    Complete(ipv6::Address),
+    WithContext(&'a [u8]),
+    Elided,
+    Reserved,
```

Comment:
The `WithContext` is defined in the [RFC](https://datatracker.ietf.org/doc/html/rfc6282#section-3.1.1), however I don't know how to really use it.
Instead of returning `Malformed` I added the `NotSupported` error. This way, users will know that the behaviour is defined in some standard or RFC, but that smoltcp just does not support it yet. I think this can reduce debug time for users, since we already confirm that there is well-defined behaviour, just not supported.